src/core/gatherer.ts

```diff
@@ -52,6 +57,9 @@ export class Gatherer {
     });
     const page = await context.newPage();
     const client = await context.newCDPSession(page);
+    if (networkConditions) {
+      await client.send('Network.emulateNetworkConditions', networkConditions);
```

Comment:
This session is only for page level and wont work for new tabs/windows. You need to use something like this 
https://github.com/elastic/synthetics/blob/a621b530a9559538c5d52eefbaabb8c9c9e62c8e/src/core/gatherer.ts#L67

I would move that code inside and reuse it for useragent and network throttling.

Reply:
`Network.emulateNetworkConditions` is only available on the browser session, not the page session. So I cannot set the the network conditions on the browser CDP session. To mitigate this and ensure it works on popups, I've added https://github.com/elastic/synthetics/pull/383/commits/ec3816fde03c78b328ffdd4efb34b5a2d901e17a which ensures that network conditions are set on new pages and adds a test for popups.

Reply:
Interesting, Sure that page event workaround seems to be fine.